acrobat/blocks/app-banner/app-banner.js

```diff
@@ -0,0 +1,130 @@
+import { setLibs } from '../../scripts/utils.js';
+
+const miloLibs = setLibs('/libs');
+/**
+ * Reusing from express codebase,
+ * can be replaced once Milo provides an implementation
+ * @returns mobileOS
+ */
+function getMobileOperatingSystem() {
+  const userAgent = navigator.userAgent || navigator.vendor || window.opera;
+
+  // Windows Phone must come first because its UA also contains "Android"
+  if (/windows phone/i.test(userAgent)) {
+    return 'Windows Phone';
+  }
+
+  if (/android/i.test(userAgent)) {
+    return 'Android';
+  }
+
+  if (/iPad|iPhone|iPod/.test(userAgent) && !window.MSStream) {
+    return 'iOS';
+  }
+
+  return 'unknown';
+}
+
+let link = '';
+let mobileOS = getMobileOperatingSystem();
+
+function closeBanner() {
+  const appBanner = document.querySelector('.app-banner');
+  document.body.classList.remove('has-app-banner');
+  appBanner.remove();
+}
+
+function openLink() {
+  window.open(link, '_blank');
+  closeBanner();
+}
+
+const { createTag } = await import(`${miloLibs}/utils/utils.js`);
+
+function getDecoratedBannerRight(children) {
+  const openText = children[3].textContent.trim();
+  const appBannerRight = createTag('div', { class: 'app-banner-right' });
+  const openButton = createTag('button', { class: 'app-banner-button', role: 'text', 'aria-label': openText }, openText);
+  appBannerRight.addEventListener('click', openLink);
+  appBannerRight.append(openButton);
+  return appBannerRight;
+}
+
+function getDecoratedAppDetails(children) {
+  let rating = 0;
+  let reviews = 0;
+
+  if (mobileOS === 'Android') {
+    rating = children[5].textContent.trim();
+    reviews = children[6].textContent.trim();
+  } else if (mobileOS === 'iOS') {
+    rating = children[8].textContent.trim();
+    reviews = children[9].textContent.trim();
+  }
+
+  const appTitle = createTag('h5', { class: 'app-banner-title' }, children[0].textContent.trim());
+  const appDesc = createTag('p', { class: 'app-banner-description' }, children[2].textContent.trim());
+  const appStars = createTag('div', { class: 'app-banner-stars', role: 'text', 'aria-label': `Average rating ${rating} stars` });
+
+  rating = rating > 5 ? 5 : rating;
+  rating = rating < 0 ? 0 : rating;
```

Comment:
One-liner alternative. Not sure which is faster. 
```
rating = Math.min(Math.max(rating, 0), 5)
```

Reply:
Not sure :) but 2 lines are more readable.